python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -41,7 +42,7 @@ def _getName(obj):
 
 _imported_GangaList = None
 
-do_not_copy = ['_index_cache', '_parent', '_registry', '_data', '_read_lock', '_write_lock', '_proxyObject']
+do_not_copy = ['_index_cache', '_index_cache_dict', '_parent', '_registry', '_data', '_data_dict', '_read_lock', '_write_lock', '_proxyObject']
```

Comment:
Well the point of the `memo` argument in `__deepcopy__` is exactly to prevent infinite loops. Until now we weren't using is properly but this PR adds the items correctly (I think).

I had assumed that `__dict__` would contain both the getter and the dict but now I think about it, the setter is a property of the class, not the instance. This means that I can remove `_data` and `_index_cache` from the `do_not_copy`. I will do so.